users_ldap_mail/users_ldap_model.py

```diff
@@ -60,18 +57,26 @@ def get_ldap_dicts(self, cr, ids=None):
         """, args)
         return cr.dictfetchall()
 
-    def map_ldap_attributes(self, cr, uid, conf, login, ldap_entry):
-        values = super(CompanyLDAP, self).map_ldap_attributes(
-            cr, uid, conf, login, ldap_entry)
+    def get_or_create_user(self, cr, uid, conf, login, ldap_entry,
+                           context=None):
+        user_id = super(CompanyLDAP, self).get_or_create_user(
+            cr, uid, conf, login, ldap_entry, context)
+
         mapping = [
             ('name', 'name_attribute'),
             ('email', 'mail_attribute'),
         ]
+        values = {}
         for value_key, conf_name in mapping:
             try:
                 if conf[conf_name]:
                     values[value_key] = ldap_entry[1][conf[conf_name]][0]
             except KeyError:
-                _log.warning('No LDAP attribute "%s" found for login  "%s"' % (
-                    conf.get(conf_name), values.get('login')))
-        return values
+                _logger.warning(
+                    'No LDAP attribute "%s" found for login  "%s"'
+                    % (conf.get(conf_name), values.get('login')))
+
+        new_user = self.pool['res.users'].browse(cr, SUPERUSER_ID, user_id)
```

Comment:
The previous code was overriding map_ldap_attributes, but the error I am fixing is because attributes like 'email' cause some kind of a cache error when added to the login dictionary and calling get_or_create_user. My fix is to override get_or_create_user instead and then modify the user. I used superuser, because that is the way the browse of res.users to get the user object is done in res.company.ldap. I hope this explains.